oracle-uri-proposal.md

```diff
@@ -0,0 +1,196 @@
+# Oracle Event URI proposal
+
+This is a proposal for a _Uniform Resource Identifier_ (URI) format for oracle _event identifiers_ (event id).
+We take URI to mean a character string that uniquely defines a resource without implying its location or how to access it.
+From the URI style event id, a client can determine all the event outcomes attach semantics to each.
+
+## Motivation
+
+When executing a _Discreet Log Contract_ two parties agree on an event and a proportion of coins each party will receive for each event outcome.
+In addition they must also agree on an _oracle_ who will cryptographically attest to the outcome of an event.
+To do this correctly, the oracle and the two parties have precisely the same understanding of the event.
+
+A single URI string that uniquely identifies an event has the following benefits:
+
+1. Copy pasteable: A user can easily copy paste the event id into a chat app/DLC software/browser address bar.
+2. Trivially serializable: It is easy to byte encode the event id for signing purposes.
+3. Human readable: A user who is sufficiently familiar with event namespace does not need software to understand the event.
+
+## Specification
+
+An event identifier (_event id_) is a relative URI reference (a URI without the scheme and authority) with a query string.
+The path of the URI is interpreted as a description of an event in time and space.
+The query string is interpreted as an _measurement_ and describes a certain fact that will be observed by an oracle on the object described by the path.
+
+If applicable, the path should start with the human organization responsible for the event.
+If it is a price then the name of the exchange where the price is being measured e.g. `binance` or if it's a sporting competition then the name of the competition e.g. `NBA`.
```

Comment:
I'm a little worried about capitalization

Reply:
I think that proper nouns should be in capitals. So `Binance` and `NBA`. What do you think?